storage/src/cache/cachedfile.rs

```diff
@@ -318,19 +318,21 @@ impl BlobCache for FileCacheEntry {
         let mut total_size = 0;
         let mut start = 0;
         while start < pending.len() {
-            let mut end = start + 1;
-            while end < pending.len() && pending[end].id() == pending[end - 1].id() + 1 {
+            let mut end = start;
+
+            while end < pending.len() - 1 && pending[end + 1].id() == pending[end].id() + 1 {
                 end += 1;
             }
 
             // Find a range with continuous chunk id
             let blob_offset = pending[start].compress_offset();
             let blob_end = pending[end].compress_offset() + pending[end].compress_size() as u64;
```

Comment:
So it seems that the only change that we need is to fix this line here, to make sure `end` is exclusive:
```
let blob_end = pending[end - 1].compress_offset() + pending[end - 1].compress_size() as u64;
```

Looks like a typo of the original code where one forgets that `end` is exclusive. Worth a comment at line 321 to remind readers that `end` is exclusive while `start` is inclusive.

Reply:
Yes. It might be another way to address this. But I prefer current PR since it's more straightforward to represent the 
 intention and no need to add a comment line to mitigate the gap between code author and readers. Besides for L348,` take(end)` where the pending is not that long is really burning the brain 😥

Reply:
Yes. It might be another way to address this. But I prefer current PR since it's more straightforward to represent the
intention and no need to add a comment line to mitigate the gap between code author and readers. Besides for L348, take(end) where the pending is not that long is really burning the brain 😥

Reply:
aaa

Reply:
IMO, it doesn't make the code easier to understand by pivoting from one way (exclusive ending) to another (inclusive ending). OTOH, a comment stating that we have either `[start, end)` or `[start, end]` makes it clear how the code is intended to be written. W/o such a reminder, even @jiangliu mistook It when he wrote the code in the first place.

Reply:
Ture. But I am inspired by the PR https://github.com/dragonflyoss/image-service/pull/626 merged this morning.  It seems that `BlobCache::prefetch_range` and `BlobObject::prefetch_chunks` have very similar logics. So I am trying to remove the duplicated one `prefetch_range`. There are several advantages:

1. Make code neat 
2. prefech_chunks has more sophisticated logic
3. Simplify choice of using what to do backend prefetch

I will raise a PR later and if the new one is not accepted, let's do what you are suggesting with this PR

Reply:
I don't object to changing the exclusive ending to an inclusive end. But I don't agree that change would make the code any way cleaner/clearer. We need a comment to remind us how `end` is intended to be, either exclusive or inclusive. And that is how the code becomes clearer.

Reply:
If we want to have similar logics in #626 (which I have to admit that I didn't take a closer look at), please state it clearly in the commit message. Nevertheless, I still think a comment to state that we have either `[start, end)` or `[start, end]` is necessary here.

Reply:
Sure.